// Copyright 2022 Google LLC

// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at

//     https://www.apache.org/licenses/LICENSE-2.0

// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// Defines messages for normalizing and de-normalizing variables.

syntax = "proto3";

package smart_buildings.smart_control.proto;

import "google/protobuf/timestamp.proto";

// Stats for normalizing continuous random variables.
message ContinuousVariableInfo {
  // Unique ID of the random variable.
  string id = 1;
  // Start and end times of the sample.
  google.protobuf.Timestamp sample_start = 2;
  google.protobuf.Timestamp sample_end = 3;
  // Number of examples in the sample.
  int32 sample_size = 4;
  // Sample stats:
  float sample_variance = 5;
  float sample_mean = 6;
  float sample_median = 7;
  float sample_maximum = 8;
  float sample_minimum = 9;
}
